### ISSUE TITLE
inner_hits doesn't work inside indices query

### ISSUE THREAD

**Issue by unknownlighter** [elastic/elasticsearch#26133]
```
PUT test
{
  "mappings": {
    "doc": {
      "properties": {
        "comments": {
          "type": "nested"
        }
      }
    }
  }
}

PUT test/doc/1?refresh
{
  "title": "Test title",
  "comments": [
    {
      "author": "kimchy",
      "text": "comment text"
    },
    {
      "author": "nik9000",
      "text": "words words words"
    }
  ]
}
```

This query works fine and returns inner_hits in both ES 1.7 and 5.5:
```
POST test/_search
{
  "query": {
    "nested": {
      "path": "comments",
      "query": {
        "match": {"comments.text" : "words"}
      },
      "inner_hits": {} 
    }
  }
}
```

This query works differently:
```
POST test/_search
{
  "query": {
    "indices": {
      "indices": [
        "test"
      ],
      "query": {
        "nested": {
          "path": "comments",
          "query": {
            "match": {
              "comments.text": "words"
            }
          },
          "inner_hits": {}
        }
      }
    }
  }
}
```

ES 1.7 result:
```
"hits": {
      "total": 1,
      "max_score": 1.2171685,
      "hits": [
         {
            "_index": "test",
            "_type": "doc",
            "_id": "1",
            "_score": 1.2171685,
            "_source": {
               "title": "Test title",
               "comments": [
                  {
                     "author": "kimchy",
                     "text": "comment text"
                  },
                  {
                     "author": "nik9000",
                     "text": "words words words"
                  }
               ]
            },
            "inner_hits": {
               "comments": {
                  "hits": {
                     "total": 1,
                     "max_score": 1.2171685,
                     "hits": [
                        {
                           "_index": "test",
                           "_type": "doc",
                           "_id": "1",
                           "_nested": {
                              "field": "comments",
                              "offset": 1
                           },
                           "_score": 1.2171685,
                           "_source": {
                              "author": "nik9000",
                              "text": "words words words"
                           }
                        }
                     ]
                  }
               }
            }
         }
      ]
   }
```

ES 5.5 result:
```
"hits": {
    "total": 1,
    "max_score": 0.9651416,
    "hits": [
      {
        "_index": "test",
        "_type": "doc",
        "_id": "1",
        "_score": 0.9651416,
        "_source": {
          "title": "Test title",
          "comments": [
            {
              "author": "kimchy",
              "text": "comment text"
            },
            {
              "author": "nik9000",
              "text": "words words words"
            }
          ]
        }
      }
    ]
  }
```

As you see ES 1.7 results contains inner_hits, but ES 5.5 doesn't

Tested ES versions:
```
"version": {
    "number": "5.5.0",
    "build_hash": "260387d",
    "build_date": "2017-06-30T23:16:05.735Z",
    "build_snapshot": false,
    "lucene_version": "6.6.0"
  }
```
```
"version": {
      "number": "1.7.3",
      "build_hash": "05d4530971ef0ea46d0f4fa6ee64dbc8df659682",
      "build_timestamp": "2015-10-15T09:14:17Z",
      "build_snapshot": false,
      "lucene_version": "4.10.4"
   }
```

**Comment by costin** [comment]
I've been able to reproduce the issue (thanks for the nice instructions) and it looks like a bug related to index queries.
I think that, index queries being [deprecated](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-indices-query.html) (in 5.x and removed in 6), they somehow exclude the nested documents altogether from the results, hence why inner hits are empty (basically it doesn't matter if they are specified or not).

I'll leave @martijnvg comment more on this; in the meantime the workaround (as indicated in the original issue) and my recommendation is to specify the indices in the URL (which looks like it already does) instead of using index queries. 
Even if this bug is fixed, upgrading from 5.x (whenever that will be) will require the queries to be rewritten again...

Cheers,

**Comment by martijnvg** [comment]
@unknownlighter Thanks for reporting this bug. I'll merge a fix soon for this. Please note like @costin said, that you will have migrate away from the `indices` query at some point, because this query will no longer exist in 6.0 and up.

**Comment by martijnvg** [comment]
Fixed via #26138
